### PR TITLE
feat: add catalog/schema/table count as catalog metrics

### PR DESCRIPTION
**Pull request by sunng87**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This patch adds catalog/schema/table count as metrics.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/catalog/src/local/memory.rs [hunk 1]
```diff
@@ -86,6 +87,7 @@ impl CatalogManager for MemoryCatalogManager {
                 catalog: &request.catalog,
                 schema: &request.schema,
             })?;
+        increment_gauge!(crate::metrics::METRIC_CATALOG_MANAGER_TABLE_COUNT, 1.0);
```
[the remark this answers is not in the document]
Comment (sunng87) [reply src/catalog/src/local/memory.rs:90]: Good idea. Let me add a label for that. Do you think we need the granularity of schema?

[the remark this answers is not in the document]
Comment (fengjiachun) [reply src/catalog/src/local/memory.rs:90]: Perhaps also need `schema`.

[the remark this answers is not in the document]
Comment (sunng87) [reply src/catalog/src/local/memory.rs]: I added db as consistent with previous http/mysql layer metrics.

File: src/catalog/src/local/manager.rs [hunk 2]
```diff
@@ -513,7 +526,14 @@ impl CatalogManager for LocalCatalogManager {
...
         let mut sys_table_requests = self.system_table_requests.lock().await;
         sys_table_requests.push(request);
-
+        increment_gauge!(
+            crate::metrics::METRIC_CATALOG_MANAGER_TABLE_COUNT,
+            1.0,
+            &[(
+                crate::metrics::METRIC_DB_LABEL,
+                build_db_string(&catalog_name, &schema_name)
+            )],
```
Comment (fengjiachun) [inline src/catalog/src/local/manager.rs:535]: How about providing a method to return a (key, value) pair?

**Review by fengjiachun — APPROVED** [review]

File: src/catalog/src/local/manager.rs (hunk 2 above)
Comment (sunng87) [reply src/catalog/src/local/manager.rs]: resolved

**Pull request merged**